backend/services/implementations/__tests__/testService.test.ts

```diff
@@ -29,18 +34,28 @@ describe("mongo testService", (): void => {
     userService.getUserById = jest.fn().mockReturnValue(mockAdmin);
     const res = await testService.createTest(mockTest);
 
-    expect(res.id).not.toBeNull();
-    expect(res).toMatchObject({
-      ...mockTest,
-      questions: res.questions,
-      admin: mockAdmin,
-      id: res.id,
-    });
+    assertResponseMatchesExpected(mockTest, res);
   });
 
   it("createTest invalid admin userId", async () => {
     await expect(async () => {
       await testService.createTest(mockTest);
     }).rejects.toThrowError(`userId ${mockTest.admin} not found`);
   });
+
+  it("getTestById", async () => {
+    userService.getUserById = jest.fn().mockReturnValue(mockAdmin);
+    const test = await testService.createTest(mockTest);
+    const res = await testService.getTestById(test.id);
```

Comment:
we could use `const test = await MgTest.create(mockTest);` instead to avoid dependency on our own service but either way works

Reply:
we can leave it for now i think

Reply:
+1 @carissa-tang, let's try and keep tests modular and avoid dependencies!